controllers/update.go

```diff
@@ -0,0 +1,154 @@
+			dep.Spec.Template.Spec.Volumes[0].VolumeSource = registry.GetDevfileRegistryVolumeSource(cr)
+			needsUpdating = true
+		}
+	}
+	if needsUpdating {
+		log.Info("Updating the DevfileRegistry deployment")
+		return r.Update(ctx, dep)
+	}
+	return nil
+}
+
+// updateRoute checks to see if any of the fields in an existing devfile index route needs updating
+func (r *DevfileRegistryReconciler) updateRoute(ctx context.Context, cr *registryv1alpha1.DevfileRegistry, route *routev1.Route) error {
+	needsUpdating := false
+
+	// Check to see if TLS fields were updated
+	if registry.IsTLSEnabled(cr) {
+		if route.Spec.TLS == nil {
+			route.Spec.TLS = &routev1.TLSConfig{Termination: routev1.TLSTerminationEdge}
+			needsUpdating = true
+		}
+	} else {
+		if route.Spec.TLS != nil {
+			route.Spec.TLS = nil
+			needsUpdating = true
+		}
+	}
+
+	if needsUpdating {
+		return r.Update(ctx, route)
+	}
+	return nil
+}
+
+// updateIngress checks to see if any of the fields in an existing ingress resouorce need to be updated
+func (r *DevfileRegistryReconciler) updateIngress(ctx context.Context, cr *registryv1alpha1.DevfileRegistry, hostname string, ingress *v1beta1.Ingress) error {
+	needsUpdating := false
+	// Check to see if TLS fields were updated
+	if registry.IsTLSEnabled(cr) {
+		if len(ingress.Spec.TLS) == 0 {
+			// TLS was toggled on, so enable it in the ingress spec
+			ingress.Spec.TLS = []v1beta1.IngressTLS{
+				{
+					Hosts:      []string{hostname},
+					SecretName: cr.Spec.TLS.SecretName,
+				},
+			}
+			needsUpdating = true
+		}
+		if ingress.Spec.TLS[0].SecretName != cr.Spec.TLS.SecretName {
+			// TLS secret name was updated, so update it in the ingress spec
+			ingress.Spec.TLS[0].SecretName = cr.Spec.TLS.SecretName
+			needsUpdating = true
+		}
+	} else {
+		if len(ingress.Spec.TLS) > 0 {
+			// TLS was disabled, so disable it in the ingress spec
+			ingress.Spec.TLS = []v1beta1.IngressTLS{}
+			needsUpdating = true
+		}
+	}
+
+	// Check to see if the ingress domain was updated
+	if ingress.Spec.Rules[0].Host != hostname {
+		ingress.Spec.Rules[0].Host = hostname
+
+		// If TLS is enabled, need to update the hostname there too
+		if registry.IsTLSEnabled(cr) {
+			ingress.Spec.TLS[0].Hosts = []string{hostname}
+		}
+		needsUpdating = true
+	}
+
+	if needsUpdating {
+		return r.Update(ctx, ingress, &client.UpdateOptions{})
+	}
+
+	return nil
+}
+
+// deletePVCIfNeeded deletes the PVC for the devfile registry if one exists and if persistent storage was disabled
+func (r *DevfileRegistryReconciler) deleteOldPVCIfNeeded(ctx context.Context, cr *registryv1alpha1.DevfileRegistry) error {
+	// Check to see if a PVC exists, if so, need to clean it up because storage was disabled
+	if !registry.IsStorageEnabled(cr) {
+		pvc := &corev1.PersistentVolumeClaim{}
+		err := r.Get(ctx, types.NamespacedName{Name: registry.PVCName(cr.Name), Namespace: cr.Namespace}, pvc)
+		if err != nil && errors.IsNotFound(err) {
```

Comment:
```
if (condition1) {
...
} else if (condition2) {
...
}

Set1 satisfies condition1, set2 satisfies condition2, but se1 is subset of set2
```

IMO, the above if-else logic is a little bit weird, it would be better we use
```
if err != nil {
   if errors.IsNotFound(err) {
       // can't get resource -> resource does not exist   
   } else {
       // can't get resource -> resource exist with getting error
   }
} else {
   // can get resource
}

```
But the current code doesn't affect the functionality, so it's fine for now.

Reply:
Yeah, agreed.

Reply:
Updated.